File: src/perl/hello_name.pl
# Perl
# Asks for the users name then says hello
# to run: perl -e perl.pl

use strict; # disables some features (good practice but not needed)
use warnings; # turns on all warnings (good practice but not needed)

sub main{
	# prompt
	print "Enter your name: ";
	
	# get user input
	my $name = <STDIN>;

	# remove the last char (new line)
	chomp $name;

	# say hello
	print "Hello $name!\n";


}
main();

File: src/go/hello_name.go
/*
* go-lang
* asks for the users name then says hi
* run with: go run go.go
*/

package main

import ("fmt") // standard text formatting

func main(){
	// promtp
	fmt.Printf("Enter your name: ")

	// get user input
	var name string
	fmt.Scanln(&name)

	// say hello
	fmt.Printf("Hello %s!\n", name)
}

